plugins/pluginpool/scripts/doctor.py

```
@@ -0,0 +1,225 @@
+    if (root / "Cargo.toml").exists():
+        ecosystems.append("cargo")
+    if (root / "go.mod").exists():
+        ecosystems.append("go")
+    return ecosystems
+
+
+def run_audit(name: str) -> tuple[str | None, str | None]:
+    command = COMMANDS[name]
+    if shutil.which(command[0]) is None:
+        return None, "skipped: tool not installed"
+    result = subprocess.run(command, text=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)